pkg/util/kube/pvc_pv.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright The Velero Contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kube
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1api "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const (
+	waitInternal = 2 * time.Second
+)
+
+// DeletePVCIfAny deletes a PVC by name if it exists, and log an error when the deletion fails
+func DeletePVCIfAny(ctx context.Context, pvcGetter corev1client.PersistentVolumeClaimsGetter, pvcName string, pvcNamespace string, log logrus.FieldLogger) {
```

Comment:
Just my feeling, there is no need to convert the unstructured client into a client that works for specific resources.

Reply:
Done, changed it to `corev1client.CoreV1Interface`